### PR TITLE
Update to Static.jl v0.8

### PR DESCRIPTION
**Pull request by oschulz**
Closes #99 and #111 .

Since StaticInt is no longer an `Integer` in Static v0.8, we need to jump through a few hoops.

Drops support for older versions of Static.jl.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -6,8 +6,12 @@ on:
   push:
     branches:
       - master
+      - main
+      - dev
+      - 'releases/**'
```
Comment (theogf) [inline .github/workflows/CI.yml:11]: I don't think you need those.

File: Project.toml [hunk 2]
```diff
@@ -48,6 +48,6 @@ NaNMath = "0.3, 1"
 PrettyPrinting = "0.3, 0.4"
 Reexport = "1"
 SpecialFunctions = "2"
-Static = "0.5, 0.6"
+Static = "0.5, 0.6, 0.8"
```
Comment (theogf) [inline Project.toml:51]: Given your changes, shouild we just drop 0.5, 0.6 (and also what about 0.7)

File: src/combinators/power.jl [hunk 3]
```diff
@@ -76,7 +85,7 @@ end
 end
 
 @inline function logdensity_def(
-    d::PowerMeasure{M,Tuple{Base.OneTo{StaticInt{N}}}},
+    d::PowerMeasure{M,Tuple{StaticOneTo{N}}},
```
Comment (theogf) [inline src/combinators/power.jl:88]: StaticOneTo might be undefined no?

File: src/static.jl [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    # ranges, some operations that automatic differentiation requires do fail
+    # on such instances of `Fill` (e.g. `reshape` from dynamic to static size).
+    # So need to use standard ranges for the axes for now:
+    dyn_axs = map(_dynamic, axs)
+    FillArrays.Fill(x, dyn_axs)
+end
+
+
+"""
+    MeasureBase.dslength(x)::IntegerLike
```
Comment (theogf) [inline src/static.jl:67]: The `ds_` naming is a bit hard to follow. Maybe there a could be another convention?

File: Project.toml (hunk 2 above)
Comment (oschulz) [reply Project.toml:51]: 0.7 we can't support, static numbers aren't even `Real`s there. :-(

File: src/static.jl (hunk 4 above)
Comment (oschulz) [reply src/static.jl:67]: I thought about `maybestatic_length` but that's a bit of a mouthful. :-)

Any suggestions?

File: Project.toml (hunk 2 above)
Comment (oschulz) [reply Project.toml:51]: I definitely want to keep 0.5 and 0.6 support for a little bit longer, though.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (oschulz) [reply .github/workflows/CI.yml:11]: I can take that out though, if you prefer?

File: src/combinators/power.jl (hunk 3 above)
Comment (oschulz) [reply src/combinators/power.jl:88]: StaticOneTo is gone now.

File: src/combinators/power.jl [hunk 5]
```diff
@@ -1,58 +1,67 @@
...
+"""
+    struct PowerMeasure{M,...} <: AbstractProductMeasure
 
-export PowerMeasure
+A power measure is a product of a measure with itself. The number of elements in
+the product determines the dimensionality of the resulting support.
 
+Note that power measures are only well-defined for integer powers.
+
+The nth power of a measure μ can be written μ^x.
```
Comment (cscherrer) [inline src/combinators/power.jl:13]:
```suggestion
The nth power of a measure μ can be written μ^n.
```

File: .github/workflows/CI.yml (hunk 1 above)
Comment (oschulz) [reply .github/workflows/CI.yml]: I removed it, maybe it was a bit overkill :-)

File: src/combinators/power.jl [hunk 6]
```diff
@@ -1,58 +1,67 @@
...
+"""
 struct PowerMeasure{M,A} <: AbstractProductMeasure
     parent::M
     axes::A
 end
 
+dslength(μ::PowerMeasure) = prod(dssize(μ))
+dssize(μ::PowerMeasure) = map(dslength, μ.axes)
```
[the remark this answers is not in the document]
Comment (oschulz) [reply src/combinators/power.jl:21]: It stands for "dynamic or static". As @theogf remarked, probably not an ideal naming convention. Any ideas?

File: src/static.jl (hunk 4 above)
Comment (cscherrer) [reply src/static.jl:46]: This is just internal, right? If end users won't need to type it in the REPL, I think I'd lean toward making it more descriptive, even if it's longer. Most important for people reading the code to understand it

Comment (oschulz) [reply src/static.jl]: Yes, it'll be mostly internal, though packages that use deeper measure features may need them. I've changed to the long names now.

**Comment by cscherrer** [comment]
Just to be sure I understand, it seems like static floats are working again with our interface, but ints are not, so you added special cases and dispatch patterns to handle this. Is that right?

**Comment by oschulz** [comment]
Ok now @cscherrer ?

**Comment by cscherrer** [comment]
We should still check MeasureTheory tests locally. Latest MT has
```toml
Static = "0.5, 0.6"
```
Does that mean this PR is technically a breaking change?

**Review by cscherrer — APPROVED** [review]

**Comment by oschulz** [comment]
> Ok in that case let's go with this and then MT can catch up

I'll see if I can do the Distributions extension soon, then MT can be updated in one go, maybe.
